primer/brand
# Prevent page scrolling in AnchorNav open state

## Summary

Follow up to #133 

🔗 [Try it in Storybook](https://primer-ac8ed69e81-26139705.drafts.github.io/storybook/?path=/story/components-anchornav--playground)

Prevents background scrolling when the menu is open on narrow viewports.

## List of notable changes:



- *added* new scroll-prevention feature 
- *removed* the major layout shift on menu open. There is still a little bit due to scrollbars disappearing however.
- *fixed* bug where `Anchor--stuck` class wasn't getting applied correctly in Storybook canvas mode 
- *fixed* a visual bug when AnchorNav overlays on top of SubdomainNavBar but is short on covering it completely


## What should reviewers focus on?

- Try it out!

## Steps to test:



1. Open the stories in a narrow viewport / mobile device
1. Open the menu
1. Verify you can't scroll the background


## Contributor checklist:


## Reviewer checklist:


## Screenshots:

## Scroll-prevention


https://user-images.githubusercontent.com/13340707/206769797-ffc31d73-56b6-4f46-a301-bd6c412a5211.mov


> Please try to provide before and after screenshots or videos


## SubdomainNavBar overlay bug fixed
<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

![double border](https://user-images.githubusercontent.com/13340707/206769587-c5ebbac3-c523-4274-923f-5dee2388be14.png)


 </td>
<td valign="top">


![single border and shadow showing](https://user-images.githubusercontent.com/13340707/206769620-f4e24ea6-4c13-433f-a17d-1696f5e70aa9.png)

</td>
</tr>
</table>
